skrub/_agg_joiner.py

```diff
@@ -0,0 +1,592 @@
+
+        Otherwise, if main_key is a list of keys, the target will be
+        aggregated using each key separately, then each aggregation of
+        the target will be joined on the main table.
+
+    agg_ops : str or list of str, default=None
+        Aggregation operations to perform on the auxiliary table.
+
+        numerical : {"sum", "mean", "std", "min", "max", "hist", "value_counts"}
+
+        categorical : {"mode", "count", "value_counts"}
+
+        If set to None, ['mean', 'mode'] will be used.
+
+    suffixes : list of str, default=None
+        The suffixes that will be add to each table columns in case of
+        duplicate column names, similar to `("_x", "_y")` in :obj:`~pandas.merge`.
+
+        If set to None, we will use the table index, by looking at their
+        order in 'tables', e.g. for a duplicate columns: price (main table),
+        price_1 (auxiliary table 1), price_2 (auxiliary table 2), etc.
+
+    See Also
+    --------
+    :class:`AggJoiner` :
+        Aggregates auxiliary dataframes before joining them
+        on the base dataframe.
+    :class:`FeatureAugmenter` :
+        Augments a main table by automatically joining multiple
+        auxiliary tables on it.
+
+    Examples
+    --------
+    >>> X = pd.DataFrame({
+            "userId": [1, 1, 1, 2, 2, 2],
+            "movieId": [1, 3, 6, 1, 3, 6],
+            "genre": ["drama", "drama", "comedy", "sf", "comedy", "sf"],
+        })
+    >>> y = pd.DataFrame(dict(rating=[4.0, 4.0, 4.0, 3.0, 2.0, 4.0]))
+    >>> agg_target = AggTarget(
+            main_key=["userId"],
+            suffixes=["_user"],
+            agg_ops=["value_counts"],
+        )
+    >>> agg_target.fit_transform(X, y)
+        userId  movieId   genre  rating_2.0_user  rating_3.0_user  rating_4.0_user
+    0       1        1   drama              0.0              0.0              3.0
+    1       1        3   drama              0.0              0.0              3.0
+    2       1        6  comedy              0.0              0.0              3.0
+    3       2        1      sf              1.0              1.0              1.0
+    4       2        3  comedy              1.0              1.0              1.0
+    5       2        6      sf              1.0              1.0              1.0
+    """
+
+    def __init__(self, main_key, agg_ops=None, suffixes=None):
+        self.main_key = main_key
+        self.agg_ops = agg_ops
+        self.suffixes = suffixes
+
+    def fit(self, X, y):
+        """Aggregate the target ``y`` based on keys from ``X``.
+
+        Parameters
+        ----------
+        X : dataframe
+            Must contains the columns names defined in ``main_key``.
+        y : dataframe or array-like
+            Must have the length of ``X``, with matching indices.
+            The target can be continuous or discrete, with multiple columns.
+
+            If the target is continuous, only numerical operations,
+            listed in NUM_OPS, are applied.
```

Comment:
```suggestion
            listed in NUM_OPS, can be applied.
```
Maybe? The way it's currently worded feels like all operations of `NUM_OPS` will be applied.

Reply:
Yes, good catch